### PR TITLE
Adding documentation for the CalculateCoverage dataflow pipeline.

### PR DESCRIPTION
**Pull request by Careyjmac**
As per @deflaux 's request, here is a simple tutorial on what the CalculateCoverage pipeline does, and how to go about running it locally on the command line.

### CODE REVIEW

**Comment by deflaux** [comment]
Nice work @Careyjmac !!!

I just have a couple nit-picky things but otherwise it looks good to me.  @mbookman anything to add?
  * sometimes there is "data set" and "dataset" or "read group set" and "ReadGroupSet"; please go ahead and make all the type names consistent
  * the API explorer links include the google analytics tracking tag, please goahead and strip those out

**Comment by Careyjmac** [comment]
Done and done.  :)

**Comment by pgrosu** [comment]
Why does the output annotation have to go into a Dataset?  Why can't I test it locally as standard output for just retrieving the results as an inquiry into the coverage of a set of ReadGroupSets for other analyses?  It feels a little cumbersome :(  Maybe for now just add a temporary public dataset id until the Annotation system is updated to accommodate such pipelining.

Thanks,
Paul

**Comment by dionloy** [comment]
Hi Paul,

All data generated by our API is encompassed within a Dataset for
permission control. While it's entirely possible to add a file based sink,
these are intended as examples for cloud-based genomics processing (and
thus writing to local files would be muddling that message a bit). Users
are welcome to branch and modify the examples for their own purposes of
course. Thanks!

File: docs/source/use_cases/analyze_reads/calculate_coverage.rst [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+.. include:: ../../includes/dataflow_setup.rst
+
+.. include:: ../../includes/dataflow_on_gce_setup.rst
+
+Create Output Dataset
+---------------------
+
+In order to run this pipeline, you must create a dataset using the Genomics API that the pipeline
+can output its AnnotationSet and Annotations to.  If you already have a dataset in which you have
+write access, you may use its id.  If not, you can do the following to create one:
```
Comment (mbookman) [inline docs/source/use_cases/analyze_reads/calculate_coverage.rst]: s/you may use its id/you may use it

File: docs/source/use_cases/analyze_reads/calculate_coverage.rst [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+    --project=YOUR-GOOGLE-CLOUD-PLATFORM-PROJECT-ID \
+    --stagingLocation=gs://YOUR-BUCKET/dataflow-staging \
+    --secretsFile=/PATH/TO/YOUR/client_secrets.json \
+    --references=1:552960:553984 \
+    --bucketWidth=1024 \
+    --inputDatasetId=10473108253681171589 \
+    --outputDatasetId=YOUR_OUTPUT_DATASET_ID
+
+This can take several minutes to run.  You can check your results by using the Genomics API
+Explorer (make sure to authorize yourself!):
```
Comment (mbookman) [inline docs/source/use_cases/analyze_reads/calculate_coverage.rst]: Remove the parenthetical note and add authorization as an explicit step as you have above.

File: docs/source/use_cases/analyze_reads/calculate_coverage.rst [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+
+#. First go to to the to the `AnnotationSets search request page <https://developers.google.com/apis-explorer/#p/genomics/v1beta2/genomics.annotationSets.search>`_ to determine what your newly created AnnotationSetId is by putting your output dataset id in the ``datasetIds`` field.
```
Comment (mbookman) [inline docs/source/use_cases/analyze_reads/calculate_coverage.rst]: s/to to the//

File: docs/source/use_cases/analyze_reads/calculate_coverage.rst [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+      ]
+     }
+    }
+   ]
+  }
+
+The following command will also calculate the mean coverage in the same manner as the previous
+command, but will use a select number of read group sets from 1000 Genomes instead of the entire
+dataset.  To do this, we must change the number of quantiles we are computing, as we now have
+less read group sets then the default requirement of 11:
```
Comment (mbookman) [inline docs/source/use_cases/analyze_reads/calculate_coverage.rst]: s/less/fewer

File: docs/source/use_cases/analyze_reads/calculate_coverage.rst [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+  }
+
+The above command lines run the pipeline over a small portion of the genome, only taking a few minutes.
+If modified to run over a larger portion of the genome or the entire genome, it may take a few hours
+depending upon how many machines are configured to run concurrently via ``--numWorkers``.
+
+To run this pipeline over a large portion of the genome:
+
+* Add ``--runner=DataflowPipelineRunner`` to run the pipeline on Google Cloud instead of locally.
+* Add  ``--numWorkers=#`` for faster processing that will shard the data.
```
Comment (mbookman) [inline docs/source/use_cases/analyze_reads/calculate_coverage.rst]: Remove the extra blank space after "Add" here.

File: docs/source/use_cases/analyze_reads/calculate_coverage.rst [hunk 6]
```diff
@@ -0,0 +1,227 @@
...
+* Add more references:
+
+  #. Use a comma-separated list to run over multiple disjoint regions.  For example to run over `BRCA1`_ and `BRCA2`_ ``--references=13:32889610:32973808,17:41196311:41277499``
+  #. Use ``--allReferences`` instead of ``--references=1:552960:557056`` to run over the entire genome.
+
+To run the pipeline on a different dataset, change the ``--inputDatasetId`` id parameter.
```
Comment (mbookman) [inline docs/source/use_cases/analyze_reads/calculate_coverage.rst]: I think "id" can be removed from this sentence.

File: docs/source/use_cases/analyze_reads/calculate_coverage.rst [hunk 7]
```diff
@@ -0,0 +1,227 @@
...
+
+To run the pipeline on a different group of read group sets, change the ``--readGroupSetIds`` id parameter.
```
Comment (mbookman) [inline docs/source/use_cases/analyze_reads/calculate_coverage.rst]: I think "id" can be removed from this sentence.

**Comment by pgrosu** [comment]
Hi Dion,

Thank you for the clarification and I totally agree, but when I noticed that it mentioned in the `calculate_coverage.rst` document in this PR that we can run it locally (without `--runner=DataflowPipelineRunner`), I interpreted that the results can also be seen locally for troubleshooting purposes - similar to how I can perform the following locally via [Readgroupsets.coveragebuckets](https://cloud.google.com/genomics/v1beta2/reference/readgroupsets/coveragebuckets/list):

```
$ curl -X GET -H "Content-Type: application/json"  https://www.googleapis.com/genomics/v1beta2/readgroupsets/CMvnhpKTFhD04eLE-q2yxnU/coveragebuckets?key=<removed>
{
 "coverageBuckets": [
  {
   "range": {
    "referenceName": "1",
    "start": null,
    "end": "249250621"
   },
   "meanCoverage": 8.670173
  },
...
$
```

I'll continue to explore more on my own via `TextIO.Write`.

Thanks,
Paul

**Comment by pgrosu** [comment]
Hi Dion,

I just thinking of this some more, and realized that coverage is not really an annotation but the result of some performed analysis and more reasonably should have its own type definition under a general processed result category.  Until then it probably should be saved to a gs bucket that has an association to the dataset.

Let me know what you think.

Thanks,
Paul

**Comment by dionloy** [comment]
That would work too. However we are treating 'annotation' as  very loose
definition for any data that can be mapped to a genomic range (either per
sample or reference). We'll see how it works out =).

**Comment by pgrosu** [comment]
Sounds good :)  It's still fun to play around, and we're lucky to be at this point in time to create these new standards for the cloud for genomic data.

Have a good one,
`p

**Comment by Careyjmac** [comment]
All of @mbookman 's requested changes have been made.

**Comment by pgrosu** [comment]
Thank you for merging, but I'm still getting a "Sorry this page does not exist yet" when trying to access the following link as noted in [line 74 of CalculateCoverage.java](https://github.com/googlegenomics/dataflow-java/blob/master/src/main/java/com/google/cloud/genomics/dataflow/pipelines/CalculateCoverage.java#L74) :(

http://googlegenomics.readthedocs.org/en/latest/use_cases/analyze_reads/calculate_coverage.html

~p

**Pull request merged**